src/biokbase/narrative/tests/test_job.py

```diff
@@ -695,16 +672,43 @@ def test_parent_children__ok(self):
         self.assertTrue(parent_job.was_terminal)
 
     def test_parent_children__fail(self):
-        parent_state = create_state_from_ee2(BATCH_PARENT)
+        parent_state = create_state_from_ee2(BATCH_PARENT, mode="state")
         with self.assertRaisesRegex(
             ValueError,
-            "Job with `batch_job=True` must be instantiated with child job instances",
+            "Must supply children when setting children of batch job parent"
         ):
             Job.from_state(parent_state)
 
-        child_jobs = [create_job_from_ee2(job_id) for job_id in BATCH_CHILDREN][1:]
+        child_jobs = [create_job_from_ee2(job_id, mode="attributes") for job_id in BATCH_CHILDREN][1:]
         with self.assertRaisesRegex(ValueError, "Child job id mismatch"):
             Job.from_state(
                 parent_state,
                 children=child_jobs,
             )
+
+    def test_get_viewer_params(self):
+        for job_id in ALL_JOBS:
+            if job_id == BATCH_PARENT:
+                continue
+
+            state = create_state_from_ee2(job_id, mode="state")
+
+            job = create_job_from_ee2(job_id, mode="state")
+            job.get_viewer_params(state)
+
+            job = create_job_from_ee2(job_id, mode="attributes")
+            job.get_viewer_params(state)
+
+        # do batch parent separately
+        # since it requires passing in child jobs
+        state = create_state_from_ee2(BATCH_PARENT, mode="state")
+        batch_children = [
+            create_job_from_ee2(job_id, mode="state")
+            for job_id in BATCH_CHILDREN
+        ]
+
+        job = create_job_from_ee2(BATCH_PARENT, mode="state", children=batch_children)
+        job.get_viewer_params(state)
+
+        job = create_job_from_ee2(BATCH_PARENT, mode="attributes", children=batch_children)
+        job.get_viewer_params(state)
```

Comment:
you could make this into a separate test (`test_get_viewer_params__batch_parent`) for clarity / ease of troubleshooting